Upgrade to gmime3

langemeijer opened the pull request:
This merge request contains all changes to dbmail to get it to work with to gmime3.

pjstevns commented on test/check_dbmail_message.c:
```diff
@@ -834,7 +813,7 @@ START_TEST(test_dbmail_message_construct)
 	const gchar *subject = "Some test";
 	const gchar *recipient = "<bar@foo.org> Bar";
-	gchar *body = g_strdup("\ntesting\n\n�����\n\n");
+	gchar *body = g_strdup("\ntesting\n\n�����\n\n");
```
something went wrong with the encoding of the file, it seems

langemeijer replied:
These two changes were accidental, and I've reverted the lines to what it was in commit e6e1eea.

I've checked all other commits, but I did not see any other accidental changes.

langemeijer commented:
Travis CI fails because it doesn't use gmime3, what can I do to fix that?

pjstevns commented:
Casper, travis basically runs 'docker build .' in de root directory of the project. The tests are ran as part of the build pipeline.

In etc/apk I added overlays for older alpine releases since we needed gmime2 and libsieve which are not present in the latest alpine repo.

It should be sufficient for now to install the correct gmime version in the Dockerfile for now. If you've used docker before, it should be obvious what you need to change.

I'm actually thrilled to finally have a public CI pipeline!

langemeijer commented:
Good news: I have the docker

langemeijer commented:
Good news: I have the docker build environment running, tests are running.
Bad news: The tests are failing. I have some work to do.

Good news: This is the week that I'm assigned time to do so!

langemeijer commented:
I had missed you review work this morning, sorry for that. I had done some work last week and pushed this this morning. I already saw that some of it overlaps. I will review your langemeijer- branch today and tomorrow and see if I had already fixed the changes you've made, and apply the ones I've missed.

langemeijer commented:
Also: gmime3 behaves differently from 2.6 in some aspects. It is hard for me to understand if only the tests will fail, or if it has real-life implications. Most imporatant: gmime2.6 reconstructing an email from an object **includes** the 'From ' envelope line if the message was created from a message string that included that.
gmime3 does not. According to gmime3 a message consists of headers and a body. A 'From ' line is not included in the object.

Also parsing email address is different. See one of the changes I made in test_internet_address_list_parse https://github.com/dbmail/dbmail/pull/3/commits/45843af6a9cd81b6cf16d8b09755ed816437951c#diff-f6d039f0980682b604db06a11e194792L271 I don't think this would break dbmail, but I'm not entirely sure.

langemeijer commented:
I've rebased this branch on the dbmail/langemeijer-upgrade-to-gmime3 branch, dropped a few of my commits that solved the same issues and then force-pushed to this branch. Still working on this branch.

langemeijer commented:
I've been ill in bed the past few days, parts of this code keep flashing through my mind. I know some of the issues and will get back on it. Anyone else is free to look at it of course, but keep in mind: it's a construction site.

CozC commented:
Already done
